src/aktualizr_repo/image_repo.cc

```diff
@@ -44,4 +50,17 @@ void ImageRepo::addImage(const boost::filesystem::path &image_path) {
   timestamp["meta"]["snapshot.json"]["version"] = snapshot["version"].asUInt();
   Utils::writeFile(repo_dir / "timestamp.json",
                    Utils::jsonToCanonicalStr(signTuf(Uptane::Role::Timestamp(), timestamp)));
+}
+
+void ImageRepo::addImage(const std::string &name, const std::string &hash, const uint64_t length) {
+  Json::Value target;
+  target["length"] = Json::UInt(length);
+  if (hash.size() == 64) {
```

Comment:
Not really elegant... Could we use `Uptane::Hash` and have too separate arguments on the commandline, like `targetsha256` and `targetsha512`, or does it make cli handling too complicated?